pkg/client/typed/discovery/restmapper.go

```diff
@@ -0,0 +1,166 @@
+package discovery
+
+import (
+	"sync"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apimachinery/registered"
+)
+
+type discoveryRESTMapper struct {
+	discoveryClient DiscoveryInterface
```

Comment:
I would actually like a RESTMapper that I could re-use for my work. So could this be structured as:

- A method on the discovery client like `ServerResources` but preserves the priority information
- A utility somewhere to filter these resources to those that are registered in the client
- A RESTMapper that takes these resources along with their respective `VersionInterfaces`

?


I just think it would be confusing to have multiple discovery rest mappers that do very different things.